pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -0,0 +1,144 @@
+	storage["prioritylevelconfigurations/status"] = priorityLevelConfigurationStatusStorage
+
+	return storage
+}
+
+func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
+	systemPreset := SystemPresetData{
+		FlowSchemas:                 bootstrap.SystemFlowSchemas(),
+		PriorityLevelConfigurations: bootstrap.SystemPriorityLevelConfigurations(),
+	}
+	// TODO: default flow-schemas and priority levels
+	return PostStartHookName, systemPreset.EnsureSystemPresetConfiguration(), nil
+}
+
+type SystemPresetData struct {
+	FlowSchemas                 []*flowcontrolv1alpha1.FlowSchema
+	PriorityLevelConfigurations []*flowcontrolv1alpha1.PriorityLevelConfiguration
+}
+
+func (d SystemPresetData) EnsureSystemPresetConfiguration() genericapiserver.PostStartHookFunc {
+	return func(hookContext genericapiserver.PostStartHookContext) error {
+		// Adding system priority classes is important. If they fail to add, many critical system
+		// components may fail and cluster may break.
+		err := wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
+			flowcontrolClientSet, err := flowcontrolclient.NewForConfig(hookContext.LoopbackClientConfig)
+			if err != nil {
+				utilruntime.HandleError(fmt.Errorf("unable to initialize client: %v", err))
+				return false, nil
+			}
+
+			for _, flowSchema := range d.FlowSchemas {
+				_, err := flowcontrolClientSet.FlowSchemas().Get(flowSchema.Name, metav1.GetOptions{})
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						_, err := flowcontrolClientSet.FlowSchemas().Create(flowSchema)
+						if err != nil && !apierrors.IsAlreadyExists(err) {
+							return false, err
+						} else {
+							klog.Infof("created system preset FlowSchema %s", flowSchema.Name)
+						}
+					} else {
+						// Unable to get the priority class for reasons other than "not found".
+						klog.Warningf("unable to get FlowSchema %v: %v. Retrying...", flowSchema.Name, err)
+						return false, nil
+					}
+				}
+			}
+			for _, priorityLevelConfiguration := range d.PriorityLevelConfigurations {
+				_, err := flowcontrolClientSet.PriorityLevelConfigurations().Get(priorityLevelConfiguration.Name, metav1.GetOptions{})
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						_, err := flowcontrolClientSet.PriorityLevelConfigurations().Create(priorityLevelConfiguration)
+						if err != nil && !apierrors.IsAlreadyExists(err) {
+							return false, err
+						} else {
+							klog.Infof("created system preset PriorityLevelConfiguration %s", priorityLevelConfiguration.Name)
+						}
+					} else {
+						// Unable to get the priority class for reasons other than "not found".
+						klog.Warningf("unable to get PriorityLevelConfiguration %v: %v. Retrying...", priorityLevelConfiguration.Name, err)
+						return false, nil
+					}
+				}
+			}
+			klog.Infof("all system flow-control settings are created successfully or already exist.")
```

Comment:
This should be at some specific `V` level.  We do not want this message appearing once per second in the default config.  I suggest `V(4)` or numerically higher.

Reply:
updated